### PR TITLE
Computer Vision Javascript client

### PR DESCRIPTION
**Pull request by brucearmstrong**
This submission has it all! Python, HTML, Javascript, CSS, whatever language Dockerfiles are written.

Main logic is in index.js. That's where we send the images to the server and process the results.

### CODE REVIEW

File: ComputerVisionServer/moedx/tracker/views.py [hunk 1]
```diff
@@ -264,6 +275,11 @@ def recognizer_add(request):
...
+    # json_ret = json.dumps(ret)
+    # return HttpResponse(json_ret)
```
Comment (ah1053) [inline ComputerVisionServer/moedx/tracker/views.py:281]: dead code

File: ComputerVisionServer/moedx/static/js/poseRenderer.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      let keypoint2 = pose[indexEnd];
+      let x2 = keypoint2[0] * renderScale;
+      let y2 = keypoint2[1] * renderScale;
+      let score2 = keypoint2[2];
+
+      if(score1 == 0 || score2 == 0) {
+          continue;
+      }
+
+      // console.log("Drawing indexStart="+indexStart+" indexEnd="+indexEnd+" ("+x1+","+y1+","+x2+","+y2+")");
```
Comment (ah1053) [inline ComputerVisionServer/moedx/static/js/poseRenderer.js:52]: dead code

File: ComputerVisionServer/moedx/static/js/index.js [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+  let width = right - x;
+  let height = bottom - y;
+
+  ctx.beginPath();
+  ctx.lineWidth = "4";
+  ctx.strokeStyle = strokeStyle;
+  if (shape == "rect") {
+    ctx.rect(x, y, width, height);
+    // ctx.fillRect(x, y, width, height);
+  } else if (shape == "circle") {
```
Comment (ah1053) [inline ComputerVisionServer/moedx/static/js/index.js:320]: the face detection circle looks nice

File: ComputerVisionServer/moedx/static/js/index.js [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+  }
+}
+
+restartProcessing();
+getServerCapabilities();
+
+function enableGpuActivities(enabled) {
+  console.log("enableGpuActivities("+enabled+")");
+  $("#button-od").prop('disabled', !enabled);
+  $("#button-pd").prop('disabled', !enabled);
```
Comment (ah1053) [inline ComputerVisionServer/moedx/static/js/index.js:145]: shouldn't button-fr added to the secret method 😄

**Review by ah1053 — APPROVED** [review]
⭐

**Review by ah1053 — COMMENTED** [review]
on a different note why we have this public ComputerVisionServer/moedx/server.crt?

**Comment by brucearmstrong** [comment]
> on a different note why we have this public ComputerVisionServer/moedx/server.crt?

For access to the webcam, the web page must be served via HTTPS, so this cert and key file are needed for testing on anything besides localhost. For testing on a private server, we use startup code like this:
`uvicorn moedx.asgi:application --host 0.0.0.0 --port 8008 --ssl-keyfile server.key --ssl-certfile server.crt --reload`

When running on an actual cloudlet, they aren't used at all. Selecting the TLS option for the port at app definition time means an HTTPS connection will be used, and certs generated by the system will be used instead, and don't need to be specified.

File: ComputerVisionServer/moedx/static/js/index.js (hunk 4 above)
Comment (brucearmstrong) [reply ComputerVisionServer/moedx/static/js/index.js:145]: FR is enabled by default, though any random person doing this process obviously won't be in the training database, and will be recognized as "Unknown". A future improvement could be to add a face training mode like the Android app has.

**Review by franklin-huang-mobiledgex — APPROVED** [review]

**Comment by franklin-huang-mobiledgex** [comment]
Super cool. Although face recognition thinks I am Lev haha

File: ComputerVisionServer/moedx/server.key [hunk 5]
```diff
@@ -1,15 +1,27 @@
...
+v3tvwZpDxCYJFtfpNewDI3g0mivtTXcvoqI1cArxE7lkjf7yDL6zzjloUAOfpLDk
+X1jFnfHHjHribAnpXaU5pbICQpP84cvsLY8oMIcCgYBRrSreDmlXe/CmG7ZWolMi
+kqZdiw817KS12a715W75pwPQMRfAi0sG0Eq301ujEjEgtSZ7+eCCLhQ9GUAgdgAZ
+pjv303VFAFt4uKFGMor0Hi3o0MJSzdDNuFG+ZU1LeG7uulmeqz5U+B5kWOY5aDzj
+5RCdst6uI//DHrnd+Hp3gQKBgQCsNgGd3WrgZI0j2zoY6XYmhecI/xyrrL7D8IsR
+NhOC9nu/giZ8kzFsLm9SN5bFNDF2ieNxGzPhqQRBZmlp1tkfmxSVO97ODV4M01Hs
+irq3Z6fsL4CVTmMYWnQUujMuR20/kVHEOagKB38chk4yjT+PGvWDD6QQuLmaqMvH
+L82NTwKBgQDNNv7Lbx9yrzvtelvo4dYoERvPt8N2m1NNGCVwlyNrN2nISZCbOKAj
+KUYZxejNYOx/qM+FN4ab2TWAugnR7rAWCVtPVt4z7MPumfOMYnUJ60tSiEL3qsDC
+vBjBOcsOHMNTvqeIy0KMkPFDKyxrhclKBovvhxWTC5lnRDwAvjC7VQ==
```
Comment (lgarner) [inline ComputerVisionServer/moedx/server.key:26]: Uh, private key?

Comment (brucearmstrong) [reply ComputerVisionServer/moedx/server.key:26]: Self-signed, just for testing.

**Pull request merged**